analyzers/tests/SonarAnalyzer.Test/TestCases/UnusedPrivateMember.CSharp9.cs

```diff
@@ -168,3 +169,121 @@ public class ClassWithPrintMembers
+            private int privateField;
+            private int PrivateProperty { get; set; }
+            private void PrivateMethod() { }
+            private ClassInstantiatedThroughReflection() { }
+            private event EventHandler PrivateEvent;
+
+            public ClassInstantiatedThroughReflection(int arg)  // Compliant - the constructor used in CreateInstance through Reflection
+            {
+            }
+
+            private class NestedType
+            {
+                private int privateField;
+            }
+        }
+
+        [DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.All)]
+        private class TypeDecoratedWithDynamicallyAccessedMembers
+        {
+            private const int PrivateConst = 42;
+            private int privateField;
+            private int PrivateProperty { get; set; }
+            private void PrivateMethod() { }
+            public TypeDecoratedWithDynamicallyAccessedMembers() { }
+            private event EventHandler PrivateEvent;
+
+            private class NestedType
+            {
+                private const int PrivateConst = 42;
+                private int privateField;
+                private int PrivateProperty { get; set; }
+                private void PrivateMethod() { }
+                private NestedType() { }
+                private event EventHandler PrivateEvent;
+            }
+        }
+
+        private class MembersDecoratedWithAttribute                             // Noncompliant
+        {
+            private const int PrivateConst = 42;                                // Noncompliant
+            [DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.NonPublicFields)] private const int PrivateConstWithAttribute = 42;
+
+            private int privateField;                                           // Noncompliant
+            [DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.NonPublicFields)] private int privateFieldWithAttribute;
+
+            private int PrivateProperty { get; set; }                           // Noncompliant
+            [DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.NonPublicProperties)] private int PrivatePropertyWithAttribute { get; set; }
+
+            private void PrivateMethod() { }                                    // Noncompliant
+            [DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.NonPublicMethods)] private void PrivateMethodWithAttribute() { }
+
+            private class NestedType { }                                        // Noncompliant
+            [DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.NonPublicNestedTypes)] private class NestedTypeWithAttribute { }
+
+            [return: DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.NonPublicMethods)]
+            private int PrivateMethodWithReturn() { return 0; } // Noncompliant
```

Comment:
The FP here is that the return type is not detected. Replace `int` with some private class a trimmer might remove.